bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/discovery/ElroConnectsBridgeDiscoveryService.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.elroconnects.internal.discovery;
+
+import static org.openhab.binding.elroconnects.internal.ElroConnectsBindingConstants.*;
+
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.elroconnects.internal.ElroConnectsBindingConstants;
+import org.openhab.binding.elroconnects.internal.devices.ElroConnectsConnector;
+import org.openhab.binding.elroconnects.internal.handler.ElroConnectsAccountHandler;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.DiscoveryService;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link ElroConnectsBridgeDiscoveryService} is used to discover K1 Hubs attached to an ELRO Connects cloud account.
+ *
+ * @author Mark Herwege - Initial Contribution
+ */
+@NonNullByDefault
+public class ElroConnectsBridgeDiscoveryService extends AbstractDiscoveryService implements ThingHandlerService {
+
+    private final Logger logger = LoggerFactory.getLogger(ElroConnectsBridgeDiscoveryService.class);
+
+    private @Nullable ElroConnectsAccountHandler accountHandler;
+
+    private volatile @Nullable ScheduledFuture<?> discoveryJob;
+
+    private static final int TIMEOUT_S = 5;
+    private static final int INITIAL_DELAY_S = 10; // initial delay for polling to allow time for login and retrieval in
+                                                   // ElroConnectsAccountHandler to complete
+    private static final int REFRESH_INTERVAL_S = 60;
+
+    public ElroConnectsBridgeDiscoveryService() {
+        super(ElroConnectsBindingConstants.SUPPORTED_CONNECTOR_TYPES_UIDS, TIMEOUT_S);
+        logger.debug("Bridge discovery service started");
+    }
+
+    @Override
+    protected void startScan() {
+        scheduler.submit(this::discoverConnectors); // If background account polling is not enabled for the handler,
```

Comment:
Since it seems to be "fire and forget", `execute` could be considered.

Reply:
Will do.